Documentation/ceph-cluster-crd.md

```diff
@@ -1319,144 +1319,72 @@ The features available from the external cluster will vary depending on the vers
-* `ROOK_EXTERNAL_FSID`: the fsid of the external Ceph cluster, it can be retrieved via the `ceph fsid` command
-* `ROOK_EXTERNAL_CEPH_MON_DATA`: is a common-separated list of running monitors IP address along with their ports, e.g: `a=172.17.0.4:3300,b=172.17.0.5:3300,c=172.17.0.6:3300`. You don't need to specify all the monitors, you can simply pass one and the Operator will discover the rest. The name of the monitor is the name that appears in the `ceph status` output.
+- Consumer cluster: The K8s cluster that will be consuming the external source cluster
 
-Now, we need to give Rook a key to connect to the cluster in order to perform various operations such as health cluster check, CSI keys management etc...
-It is recommended to generate keys with minimal access so the admin key does not need to be used by the external cluster.
-In this case, the admin key is only needed to generate the keys that will be used by the external cluster.
-But if the admin key is to be used by the external cluster, set the following variable:
+### Commands on the source Ceph cluster
 
-* `ROOK_EXTERNAL_ADMIN_SECRET`: **OPTIONAL:** the external Ceph cluster admin secret key, it can be retrieved via the `ceph auth get-key client.admin` command.
+In order to configure an external Ceph cluster with Rook, we need to extract some information in order to connect to that cluster.
 
-> **WARNING**: If you plan to create CRs (pool, rgw, mds, nfs) in the external cluster, you **MUST** inject the client.admin keyring as well as injecting `cluster-external-management.yaml`
-
-**Example**:
+1. Run the python script [create-external-cluster-resources.py](/deploy/examples/create-external-cluster-resources.py) for creating all users and keys.
 
 ```console
-export NAMESPACE=rook-ceph-external
-export ROOK_EXTERNAL_FSID=3240b4aa-ddbc-42ee-98ba-4ea7b2a61514
-export ROOK_EXTERNAL_CEPH_MON_DATA=a=172.17.0.4:3300
-export ROOK_EXTERNAL_ADMIN_SECRET=AQC6Ylxdja+NDBAAB7qy9MEAr4VLLq4dCIvxtg==
-```
-
-If the Ceph admin key is not provided, the following script needs to be executed on a machine that can connect to the Ceph cluster using the Ceph admin key.
-On that machine, run:
-
-```sh
-. deploy/examples/create-external-cluster-resources.sh
+python3 create-external-cluster-resources.py --rbd-data-pool-name <pool_name> --cephfs-filesystem-name <filesystem-name> --rgw-endpoint  <rgw-endpoint> --namespace <namespace> --rgw-pool-prefix <rgw-pool-prefix> --format bash
 ```
 
-The script will source all the necessary environment variables for you. It assumes the namespace name is `rook-ceph-external`.
-This can be changed by running the script like (assuming namespace name is `foo` this time):
-
-```sh
-ns=foo . deploy/examples/create-external-cluster-resources.sh
-```
+- `--namespace`: Namespace where CephCluster will run in the consumer cluster, for example `rook-ceph-external`
+- `--format bash`: The format of the output
+- `--rbd-data-pool-name`: The name of the RBD data pool
```

Comment:
Yes, it's required to run the python script,
It is just because of how it is designed,https://github.com/rook/rook/blob/62a31f1d0bbfacd5f5356452077121b17c1303f4/deploy/examples/create-external-cluster-resources.py#L285

I don't think any specific reason for keeping it compulsory,
Maybe @aruniiird or @leseb might have an idea why we keep it mandatory.

Reply:
We probably just expect the common scenario will use an rbd pool. That's fine if it's currently mandatory, just wanted to clarify for the doc. If it's made optional, that can be a future PR.